dali/pipeline/operators/color_space/color_space_conversion.cu

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cuda_runtime_api.h>
+#include <utility>
+#include <vector>
+#include "dali/pipeline/operators/color_space/color_space_conversion.h"
+
+namespace dali {
+
+namespace detail {
+
+template<size_t C1, size_t C2, size_t C, typename T = uint8_t>
+__global__ void SwapPackedChannelsKernel(const T *input, T *output, size_t total_pixels) {
+  static_assert(C <= 4,           "C>4 not supported");
+  static_assert(C1 < C && C2 < C, "C1 and C2 in the range 0 .. C-1");
+  static_assert(C1 != C2,         "C1 and C2 cannot be the same");
+
+  size_t idx = blockIdx.x * blockDim.x + threadIdx.x;
+  if (idx >= total_pixels) {
+    return;
+  }
+
+  const T* pixel_in = &input[idx * C];
+  T* pixel_out = &output[idx * C];
+
+  memcpy(pixel_out, pixel_in, sizeof(T)*C);
```

Comment:
I wonder if it's a good idea to use `memcpy` in a kernel.

Reply:
will fix